src/NServiceBus.Core/Transports/Msmq/MessagePump.cs

```diff
@@ -146,36 +146,41 @@ public void Start(PushRuntimeSettings limitations)
-                            receiveCircuitBreaker.Success();
-                        }
-                        catch (MessageQueueException ex)
+                        using (var tokenSource = new CancellationTokenSource())
                         {
-                            if (ex.MessageQueueErrorCode == MessageQueueErrorCode.IOTimeout)
+                            try
                             {
-                                //We should only get an IOTimeout exception here if another process removed the message between us peeking and now.
-                                return;
+                                await receiveStrategy.ReceiveMessage(inputQueue, errorQueue, tokenSource, pipeline).ConfigureAwait(false);
+                                receiveCircuitBreaker.Success();
+                            }
+                            catch (MessageQueueException ex)
+                            {
+                                if (ex.MessageQueueErrorCode == MessageQueueErrorCode.IOTimeout)
+                                {
+                                    //We should only get an IOTimeout exception here if another process removed the message between us peeking and now.
+                                    return;
+                                }
+
+                                Logger.Warn("MSMQ receive operation failed", ex);
+                                await receiveCircuitBreaker.Failure(ex).ConfigureAwait(false);
+                            }
+                            catch (OperationCanceledException)
+                            {
+                                // Intentionally ignored
+                            }
+                            catch (Exception ex)
+                            {
+                                Logger.Warn("MSMQ receive operation failed", ex);
+                                await receiveCircuitBreaker.Failure(ex).ConfigureAwait(false);
+                            }
+                            finally
+                            {
+                                concurrencyLimiter.Release();
                             }
-
-                            Logger.Warn("MSMQ receive operation failed", ex);
-                            await receiveCircuitBreaker.Failure(ex).ConfigureAwait(false);
-                        }
-                        catch (Exception ex)
-                        {
-                            Logger.Warn("MSMQ receive operation failed", ex);
-                            await receiveCircuitBreaker.Failure(ex).ConfigureAwait(false);
-                        }
-                        finally
-                        {
-                            concurrencyLimiter.Release();
                         }
-                    }, tokenSource.Token).ContinueWith(t => tokenSource.Dispose());
+                    }, CancellationToken.None);
```

Comment:
seems to be consistent with the rest of the file where we also pass Cts.None (e.g. https://github.com/Particular/NServiceBus/blob/MessagePumpCts/src/NServiceBus.Core/Transports/Msmq/MessagePump.cs#L72)